frontend/capstone-frontend/src/DashComponents/AppointmentsGrid.jsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+import AppointmentDetailsModal from './AppointmentDetailsModal'
+import { IoMailUnread } from "react-icons/io5"
+import { IoMailOpenOutline } from "react-icons/io5"
+
+function AppointmentsGrid({ appointments, role, onRead }) {
+    const [selectedAppointment, setSelectedAppointment] = useState(null)
+    const showModal = selectedAppointment !== null
+
+    function handleOpenModal(appointment) {
+        setSelectedAppointment(appointment)
+    }
+
+    function handleCloseModal() {
+        setSelectedAppointment(null)
+    }
+
+    return(
+        <>
+            <div className="mt-8 mx-6 grid gap-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 justify-center">
```

Comment:
Similar comment regarding if it is possible to define a class for your webapp to replace the CSS here?